`cargo test` now fails to run selected tests but continues to work for all tests

**Describe the bug**

`cargo test` still works when this is the entire command.

But when clicking on the individual test run links/button in VS Code that run `cargo test` with command-line switches to execute only the specific test or group of tests it now fails to build.

**To Reproduce**

Run an individual test, such as:
```
cargo test --package harper-core --lib -- linting::adjective_of_a::tests::correct_bad_of_an --exact --show-output
```

**Expected behavior**
Tests pass, or fail, but build at leasts succeeds to actually run the tests.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Platform**
Windsurf on macOS

**Full output**
```
 *  Executing task: cargo test --package harper-core --lib -- linting::adjective_of_a::tests::correct_bad_of_an --exact --show-output 

   Compiling harper-core v0.30.0 (/Users/hippietrail/harper/harper-core)
error[E0277]: `Rc<P>` cannot be shared between threads safely
  --> harper-core/src/patterns/mod.rs:59:41
   |
59 | #[cfg_attr(not(feature = "concurrent"), blanket(derive(Rc, Arc)))]
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^ `Rc<P>` cannot be shared between threads safely
   |
   = help: the trait `Sync` is not implemented for `Rc<P>`
note: required by a bound in `patterns::Pattern`
  --> harper-core/src/patterns/mod.rs:60:27
   |
60 | pub trait Pattern: Send + Sync {
   |                           ^^^^ required by this bound in `Pattern`
   = note: this error originates in the attribute macro `blanket` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `Rc<P>` cannot be sent between threads safely
  --> harper-core/src/patterns/mod.rs:59:41
   |
59 | #[cfg_attr(not(feature = "concurrent"), blanket(derive(Rc, Arc)))]
   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^ `Rc<P>` cannot be sent between threads safely
   |
   = help: the trait `Send` is not implemented for `Rc<P>`
note: required by a bound in `patterns::Pattern`
  --> harper-core/src/patterns/mod.rs:60:20
   |
60 | pub trait Pattern: Send + Sync {
   |                    ^^^^ required by this bound in `Pattern`
   = note: this error originates in the attribute macro `blanket` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `F` cannot be shared between threads safely
   --> harper-core/src/patterns/mod.rs:133:21
    |
133 | impl<F> Pattern for F
    |                     ^ `F` cannot be shared between threads safely
    |
note: required by a bound in `patterns::Pattern`
   --> harper-core/src/patterns/mod.rs:60:27
    |
60  | pub trait Pattern: Send + Sync {
    |                           ^^^^ required by this bound in `Pattern`
help: consider further restricting type parameter `F` with trait `Sync`
    |
135 |     F: Fn(&Token, &[char]) -> bool + std::marker::Sync,
    |                                    +++++++++++++++++++

error[E0277]: `F` cannot be sent between threads safely
   --> harper-core/src/patterns/mod.rs:133:21
    |
133 | impl<F> Pattern for F
    |                     ^ `F` cannot be sent between threads safely
    |
note: required by a bound in `patterns::Pattern`
   --> harper-core/src/patterns/mod.rs:60:20
    |
60  | pub trait Pattern: Send + Sync {
    |                    ^^^^ required by this bound in `Pattern`
help: consider further restricting type parameter `F` with trait `Send`
    |
135 |     F: Fn(&Token, &[char]) -> bool + std::marker::Send,
    |                                    +++++++++++++++++++

error[E0277]: `T` cannot be shared between threads safely
   --> harper-core/src/patterns/pattern_map.rs:68:21
    |
68  | impl<T> Pattern for PatternMap<T>
    |                     ^^^^^^^^^^^^^ `T` cannot be shared between threads safely
    |
note: required because it appears within the type `Row<T>`
   --> harper-core/src/patterns/pattern_map.rs:24:8
    |
24  | struct Row<T>
    |        ^^^
note: required because it appears within the type `PhantomData<Row<T>>`
   --> /Users/hippietrail/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/marker.rs:774:12
    |
774 | pub struct PhantomData<T: ?Sized>;
    |            ^^^^^^^^^^^
note: required because it appears within the type `smallvec::alloc::raw_vec::RawVec<Row<T>>`
   --> /Users/hippietrail/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:70:19
    |
70  | pub(crate) struct RawVec<T, A: Allocator = Global> {
    |                   ^^^^^^
note: required because it appears within the type `Vec<Row<T>>`
   --> /Users/hippietrail/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:397:12
    |
397 | pub struct Vec<T, #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global> {
    |            ^^^
note: required because it appears within the type `PatternMap<T>`
   --> harper-core/src/patterns/pattern_map.rs:17:12
    |
17  | pub struct PatternMap<T>
    |            ^^^^^^^^^^
note: required by a bound in `patterns::Pattern`
   --> harper-core/src/patterns/mod.rs:60:27
    |
60  | pub trait Pattern: Send + Sync {
    |                           ^^^^ required by this bound in `Pattern`
help: consider further restricting type parameter `T` with trait `Sync`
    |
70  |     T: LSend + std::marker::Sync,
    |              +++++++++++++++++++

error[E0277]: `T` cannot be sent between threads safely
   --> harper-core/src/patterns/pattern_map.rs:68:21
    |
68  | impl<T> Pattern for PatternMap<T>
    |                     ^^^^^^^^^^^^^ `T` cannot be sent between threads safely
    |
note: required because it appears within the type `Row<T>`
   --> harper-core/src/patterns/pattern_map.rs:24:8
    |
24  | struct Row<T>
    |        ^^^
note: required because it appears within the type `PhantomData<Row<T>>`
   --> /Users/hippietrail/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/marker.rs:774:12
    |
774 | pub struct PhantomData<T: ?Sized>;
    |            ^^^^^^^^^^^
note: required because it appears within the type `smallvec::alloc::raw_vec::RawVec<Row<T>>`
   --> /Users/hippietrail/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:70:19
    |
70  | pub(crate) struct RawVec<T, A: Allocator = Global> {
    |                   ^^^^^^
note: required because it appears within the type `Vec<Row<T>>`
   --> /Users/hippietrail/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:397:12
    |
397 | pub struct Vec<T, #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global> {
    |            ^^^
note: required because it appears within the type `PatternMap<T>`
   --> harper-core/src/patterns/pattern_map.rs:17:12
    |
17  | pub struct PatternMap<T>
    |            ^^^^^^^^^^
note: required by a bound in `patterns::Pattern`
   --> harper-core/src/patterns/mod.rs:60:20
    |
60  | pub trait Pattern: Send + Sync {
    |                    ^^^^ required by this bound in `Pattern`
help: consider further restricting type parameter `T` with trait `Send`
    |
70  |     T: LSend + std::marker::Send,
    |              +++++++++++++++++++

error[E0277]: `Rc<WordSet>` cannot be sent between threads safely
  --> harper-core/src/linting/back_in_the_day.rs:39:32
   |
39 |             if self.exceptions.matches(tail, source).is_some() {
   |                                ^^^^^^^ `Rc<WordSet>` cannot be sent between threads safely
   |
   = help: the trait `Send` is not implemented for `Rc<WordSet>`
note: required by a bound in `patterns::Pattern::matches`
  --> harper-core/src/patterns/mod.rs:60:20
   |
60 | pub trait Pattern: Send + Sync {
   |                    ^^^^ required by this bound in `Pattern::matches`
...
64 |     fn matches(&self, tokens: &[Token], source: &[char]) -> Option<NonZeroUsize>;
   |        ------- required by a bound in this associated function

error[E0277]: `Rc<WordSet>` cannot be shared between threads safely
  --> harper-core/src/linting/back_in_the_day.rs:39:32
   |
39 |             if self.exceptions.matches(tail, source).is_some() {
   |                                ^^^^^^^ `Rc<WordSet>` cannot be shared between threads safely
   |
   = help: the trait `Sync` is not implemented for `Rc<WordSet>`
note: required by a bound in `patterns::Pattern::matches`
  --> harper-core/src/patterns/mod.rs:60:27
   |
60 | pub trait Pattern: Send + Sync {
   |                           ^^^^ required by this bound in `Pattern::matches`
...
64 |     fn matches(&self, tokens: &[Token], source: &[char]) -> Option<NonZeroUsize>;
   |        ------- required by a bound in this associated function

For more information about this error, try `rustc --explain E0277`.
error: could not compile `harper-core` (lib test) due to 8 previous errors

 *  The terminal process "cargo 'test', '--package', 'harper-core', '--lib', '--', 'linting::adjective_of_a::tests::correct_bad_of_an', '--exact', '--show-output'" terminated with exit code: 101. 
 *  Terminal will be reused by tasks, press any key to close it. 
```

--- a/harper-core/src/patterns/mod.rs
+++ b/harper-core/src/patterns/mod.rs
@@ -55,8 +55,16 @@ pub use whitespace_pattern::WhitespacePattern;
 pub use word_pattern_group::WordPatternGroup;
 pub use word_set::WordSet;
 
-#[cfg_attr(feature = "concurrent", blanket(derive(Arc)))]
-#[cfg_attr(not(feature = "concurrent"), blanket(derive(Rc, Arc)))]
+#[cfg(not(feature = "concurrent"))]
+#[blanket(derive(Rc, Arc))]
+pub trait Pattern {
+    /// Check if the pattern matches at the start of the given token slice.
+    ///
+    /// Returns the length of the match if successful, or `None` if not.
+    fn matches(&self, tokens: &[Token], source: &[char]) -> Option<NonZeroUsize>;
+}
+#[cfg(feature = "concurrent")]
+#[blanket(derive(Arc))]
 pub trait Pattern: Send + Sync {
     /// Check if the pattern matches at the start of the given token slice.
     ///

